Clean up for Upstream

bgly opened the pull request.

Commit:
ibmvmc: Move Module Info

Put the module stuff at the end of the driver

Signed-off-by: Bryant G. Ly <bryantly@linux.vnet.ibm.com>

Commit:
ibmvmc: Rename functions to include ibmvmc_

This patch just renames functions to have ibmvmc_ prefix
for some functions.

Signed-off-by: Bryant G. Ly <bryantly@linux.vnet.ibm.com>

Commit:
ibmvmc: Fix CheckPatch warnings

Signed-off-by: Bryant G. Ly <bryantly@linux.vnet.ibm.com>

Commit:
ibvmc: Add new copyright author

Signed-off-by: Bryant G. Ly <bryantly@linux.vnet.ibm.com>

Commit:
ibmvmc: Remove adapter->name

We dont need to keep track of the adapter->name it isn't
used anywhere in the driver.

Signed-off-by: Bryant G. Ly <bryantly@linux.vnet.ibm.com>

Commit:
ibmvmc: Change pr_*() to dev_*() for debugging

Change where applicable all the debugs to dev_*()

Signed-off-by: Bryant G. Ly <bryantly@linux.vnet.ibm.com>

seroyer commented on ibmvmc/ibmvmc.c:
```diff
@@ -82,6 +82,8 @@ static inline void h_free_crq(uint32_t unit_address)
 		if (H_IS_LONG_BUSY(rc))
 			msleep(get_longbusy_msecs(rc));
+		else if (rc)
+			msleep(100);
```
That is longer than most long busy values.  Really, H_BUSY means retry immediately.  How do other CRQ drivers work?  If we really want a sleep in here, it should be much smaller.  More like 1ms.

bgly replied:
oh I didnt know H_BUSY meant to retry immediately. If that is the case then it is good as is. Other drivers aren't implemented as well, its just a standard if(rc) msleep(100); without the part you have up there.

Commit:
ibmvmc: Change memory barriers to lese heavyweight

PowerPc wmb() rmb() mb() barriers are very heavyweight,
the new dma_*() calls were specifically made to enforce
ordering of read and writes when accessing coherent
memory that is shared between the CPU and device.

These new calls do not guarantee ordering between
coherent and incoherent memories, but we dont need
that in the usage here.

Signed-off-by: Bryant G. Ly <bryantly@linux.vnet.ibm.com>

Commit:
ibmvmc: Styling changes

This commit just fixes up some nit styling.

Signed-off-by: Bryant G. Ly <bryantly@linux.vnet.ibm.com>

Commit:
ibmvmc: Remove unused DEFINE in header file

Signed-off-by: Bryant G. Ly <bryantly@linux.vnet.ibm.com>

Commit:
ibmvmc: Add comments for functions

Signed-off-by: Bryant G. Ly <bryantly@linux.vnet.ibm.com>

Commit:
ibmvmc: Add the Protocol for VMC Communication Doc

Signed-off-by: Bryant G. Ly <bryantly@linux.vnet.ibm.com>

bgly commented:
@seroyer This series is prob good to go, if you like everything... I plan to move ibmvmc.h to asm/ibmvmc.h and some of the stuff in there to hvcall.h. But all that I want to do kernel upstream.

seroyer commented on ibmvmc/ibmvmc.c:
```diff
@@ -420,6 +528,24 @@ static int ibmvmc_return_hmc(struct ibmvmc_hmc *hmc, bool bReleaseReaders)
+ *
+ * This command is sent by the management partition as the result of a
+ * management partition device request. It causes the hypervisor to
+ * prepare a set of data buffers for the LPM connection indicated HMC idx.
```
I would recommend search/replace LPM with HMC.  LPM was older terminology that was later changed, but the document was never updated.  I think HMC would be less confusing.

seroyer commented on ibmvmc/ibmvmc.txt:
```diff
@@ -0,0 +1,155 @@
+a system; however, when an HMC is attached to the system, the VMC interface is
+disabled by the hypervisor.
+
+Logical Partition Manager (LPM)
```
This block is no longer super accurate.  This should be rewritten to describe the function differently.  For example, NovaLink uses VMC, but isn't browser based.  (And LPM is super confusing to people used to PowerVM because it means something completely different)

seroyer commented:
One of the complaints from Greg was that he didn't see the need for a header file at all for a single .c.  Not sure if you planned to address that or not.

Commit:
ibmvmc: Fix 75 char limit for documentation

This patch fixes 75 char limit for documentation and then
also addresses rewording of LPM to Novalink.

Signed-off-by: Bryant G. Ly <bryantly@linux.vnet.ibm.com>

bgly commented:
I fixed the wording and I will address the header problem while upstreaming. The thing is most of the other character drivers or misc drivers in general have simple 1 or 2 structs and like 10 defines... In our case we have a bunch of stuff... And for the misc drivers that have a bunch of stuff they stick it in the asm directory so it would be asm/something.h

bgly commented:
Are you going to merge or should I?
